### PR TITLE
powerline-mem-segment: init at 2.4.1

### PR DESCRIPTION
**Pull request by thomasjm**
###### Motivation for this change

Add a useful segment for Powerline.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions (Ubuntu 20.04)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [N/A] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 123978` at 3cbcc4e0 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.powerlineMemSegment</li>
    <li>python39Packages.powerlineMemSegment</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-uri-literals.md">no-uri-literals</a>

URI literals are deprecated.
Near pkgs/development/python-modules/powerline-mem-segment/default.nix:20:16:
```
   |
20 |     homepage = https://github.com/mKaloer/powerline_mem_segment;
   |                ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/powerline-mem-segment/default.nix:19:0:
```
   |
19 |     description = "Segment for Powerline showing the current memory usage in percent or absolute values.";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/powerline-mem-segment/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{ lib,
+  buildPythonPackage,
+  fetchPypi,
+  psutil
+}:
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/powerline-mem-segment/default.nix:5]:
```suggestion
{ lib
, buildPythonPackage
, fetchPypi
, psutil
}:
```

File: pkgs/development/python-modules/powerline-mem-segment/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+buildPythonPackage rec {
+  pname = "powerline-mem-segment";
+  version = "2.4.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0507zw7g449zk7dcq56adcdp71inbqfdmp6y5yk4x4j7kkp6pii9";
+  };
+
+  buildInputs = [ psutil ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/powerline-mem-segment/default.nix:16]: Is that required at runtime? If so it needs to go to propagatedBuildInputs.

Comment (thomasjm) [reply pkgs/development/python-modules/powerline-mem-segment/default.nix:16]: Good catch, fixed

File: pkgs/development/python-modules/powerline-mem-segment/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  pname = "powerline-mem-segment";
+  version = "2.4.1";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0507zw7g449zk7dcq56adcdp71inbqfdmp6y5yk4x4j7kkp6pii9";
+  };
+
+  propagatedBuildInputs = [ psutil ];
+
```
Comment (jonringer) [inline pkgs/development/python-modules/powerline-mem-segment/default.nix:17]: If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.

**Review by jonringer — COMMENTED** [review]
please also check that tests are running.

File: pkgs/development/python-modules/powerline-mem-segment/default.nix (hunk 3 above)
Comment (thomasjm) [reply pkgs/development/python-modules/powerline-mem-segment/default.nix:17]: done

**Pull request merged**